apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Combobox } from '@headlessui/react';
+import { useState, useEffect, useCallback } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import ModuleView from './ModuleView';
+
+import { ReactComponent as SearchIcon } from '~/assets/SVGIcons/24px/Search.svg';
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '~/ui/Tabs';
+import { ListItem, VerticalList } from '~/ui/VerticalList';
+
+interface Props {
+    id?: string;
+    modules: [moduleName: string, code: string][];
+}
+
+const initialSelectModule = (searchParams: any, modulenames: string[]) => {
+    const paramModule = searchParams.get('module') || modulenames?.[0] || null;
+
+    if (!!paramModule && modulenames.includes(paramModule)) {
+        return paramModule;
+    } else {
+        return modulenames[0];
+    }
+};
+
+function PkgModuleViewWrapper({ id, modules }: Props) {
+    const modulenames = modules.map(([name], idx) => name);
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [query, setQuery] = useState('');
+    const [selectedModule, setSelectedModule] = useState(
+        initialSelectModule(searchParams, modulenames)
+    );
+
+    const filteredModules =
+        query === ''
+            ? modulenames
+            : modules
+                  .filter(([name]) => name.startsWith(query))
+                  .map(([name]) => name);
+
+    useEffect(() => {
+        setSearchParams({ module: selectedModule });
+    }, [selectedModule, setSearchParams]);
+
+    const submitSearch = useCallback(() => {
+        setSelectedModule((prev: string) =>
+            filteredModules.length === 1 ? filteredModules[0] : prev
+        );
+    }, [filteredModules]);
+
+    return (
+        <div className="flex flex-wrap border-0 border-y border-solid border-sui-grey-45">
+            <div className="w-full md:w-[15vw] pr-[20px]">
+                <Combobox value={selectedModule} onChange={setSelectedModule}>
+                    <div className="box-border border border-sui-grey-50 border-solid rounded-[6px] h-[34px] shadow-sm placeholder-sui-grey-65 pl-3 w-full flex my-[10px] justify-between  ml-[2px]">
+                        <Combobox.Input
```

Comment:
Last time I talked to @mystie711 about this I think the intent here was to make this a dropdown search UI rather than searching the modules inline (@mystie711 correct me if I'm wrong here).

Reply:
@mystie711 , the search functionality is demoed from second 40 in the above video. Also feel free to contact me on Slack and we can arrange a video call to get more of a live demo so you can see if there are any UX deltas.

Reply:
@Andrew47 somehow missed this comment until now. The Search UX behavior is documented here in Figma,
https://www.figma.com/file/kSD0qXCIDZA4KQVNepA5hR/02-Explorer?node-id=1566%3A109702

Please ping me on Slack if a quick huddle might be helpful.